File: NEIPG/MenuViewController.m
//
//  MenuViewController.m
//  SlideMenu
//
//  Created by Kyle Begeman on 1/13/13.
//  Copyright (c) 2013 Indee Box LLC. All rights reserved.
//

#import "MenuViewController.h"
//#import "ECSlidingViewController.h"
#import "UIViewController+ECSlidingViewController.h"
#import "PGWebViewController.h"
#import "PGDataManager.h"
//#import "ECSlidingSegue.h"
@interface MenuViewController ()
@property (strong, nonatomic) NSMutableDictionary *names;
@property (strong, nonatomic) NSMutableArray *menu;
@property (strong, nonatomic) NSMutableArray *menuKey;
@end

@implementation MenuViewController

@synthesize menu;
@synthesize menuKey;

- (id)initWithStyle:(UITableViewStyle)style
{
    self = [super initWithStyle:style];
    if (self) {
        // Custom initialization
    }
    return self;
}

- (void)viewDidLoad
{
    [super viewDidLoad];
    [self.tableView setSeparatorColor:[UIColor clearColor]];
    // Uncomment the following line to preserve selection between presentations.
    // self.clearsSelectionOnViewWillAppear = NO;
 
    // Uncomment the following line to display an Edit button in the navigation bar for this view controller.
    // self.navigationItem.rightBarButtonItem = self.editButtonItem;
    self.names =[PGDataManager getContentHeader];
    self.menuKey = [self.names objectForKey: @"key"];
    self.menu = [self.names objectForKey: @"value"];
    [self.menuKey addObject:[self.menuKey objectAtIndex:0]];
    [self.menu addObject:@"Top"];
    //did not initilaize self.menuKey
    //not ordered
    /*
    for (id key in self.names) {
        [self.menuKey addObject:key];
        NSLog(@"Key: %@, Value %@", key, [self.names objectForKey: key]);
    }*/
    
    //self.menu = [NSArray arrayWithObjects:@"BOOK 1 1", @"CNS 2", @"EBK 3", @"PODCAST 4", nil];
    //self.menuKey = [NSArray arrayWithObjects:@"BOOK", @"CNS", @"EBK", @"POD", nil];
    
    [self.slidingViewController setAnchorRightRevealAmount:200.0f];
    //self.slidingViewController.underLeftWidthLayout = ECFullWidth;
    
}

- (void)didReceiveMemoryWarning
{
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}
/*
- (void)viewWillDisappear:(BOOL)animated {
    
    [super viewWillDisappear:animated];
    [self.view endEditing:YES];
}

- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
    ECSlidingSegue *slidingSegue = (ECSlidingSegue *)segue;
    slidingSegue.skipSettingTopViewController = YES;
}
*/
#pragma mark - Table view data source

- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView
{
    // Return the number of sections.
    return 1;
}

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section
{
    // Return the number of rows in the section.
    return [self.menuKey count];
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath
{
    NSString *cellIdentifier = @"Cell";
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:cellIdentifier];
    if (cell == nil) {
        cell = [[UITableViewCell alloc] initWithStyle:UITableViewCellStyleSubtitle reuseIdentifier:cellIdentifier];
    }
    
    cell.textLabel.text = [NSString stringWithFormat:@"%@", [self.menu objectAtIndex:indexPath.row]];
    cell.textLabel.font = [UIFont systemFontOfSize:14]; //Change this value to adjust size
    //cell.textLabel.numberOfLines = 2; //Change this value to show more or less lines.
    
    return cell;
}

/*
// Override to support conditional editing of the table view.
- (BOOL)tableView:(UITableView *)tableView canEditRowAtIndexPath:(NSIndexPath *)indexPath
{
    // Return NO if you do not want the specified item to be editable.
    return YES;
}
*/

/*
// Override to support editing the table view.
- (void)tableView:(UITableView *)tableView commitEditingStyle:(UITableViewCellEditingStyle)editingStyle forRowAtIndexPath:(NSIndexPath *)indexPath
{
    if (editingStyle == UITableViewCellEditingStyleDelete) {
        // Delete the row from the data source
        [tableView deleteRowsAtIndexPaths:@[indexPath] withRowAnimation:UITableViewRowAnimationFade];
    }   
    else if (editingStyle == UITableViewCellEditingStyleInsert) {
        // Create a new instance of the appropriate class, insert it into the array, and add a new row to the table view
    }   
}
*/

/*
// Override to support rearranging the table view.
- (void)tableView:(UITableView *)tableView moveRowAtIndexPath:(NSIndexPath *)fromIndexPath toIndexPath:(NSIndexPath *)toIndexPath
{
}
*/

/*
// Override to support conditional rearranging of the table view.
- (BOOL)tableView:(UITableView *)tableView canMoveRowAtIndexPath:(NSIndexPath *)indexPath
{
    // Return NO if you do not want the item to be re-orderable.
    return YES;
}
*/

#pragma mark - Table view delegate

- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath
{
    
    NSString *identifier = [NSString stringWithFormat:@"%@", [self.menuKey objectAtIndex:indexPath.row]];
    
    
    //PGWebViewController *vc = (PGWebViewController *) self.slidingViewController.topViewController;
    UINavigationController *navController = (UINavigationController*) self.slidingViewController.topViewController;
    PGWebViewController *vc = (PGWebViewController *) [navController topViewController];
    
    //[vc reloadInputViews];
    [vc setMenuItem:identifier];
    //UIViewController *newTopViewController = [self.storyboard instantiateViewControllerWithIdentifier:identifier];
    
    //self.slidingViewController.topViewController = newTopViewController;
    
    [self.slidingViewController resetTopViewAnimated:YES];

}

@end

File: Swift Conversion/Controller/MenuViewController.swift
//
//  MenuViewController.
//  NEIPG
//
//  Created by Gourav Sharma on 11/30/16.
//  Copyright © 2016 Mobileprogrammingllc. All rights reserved.
//

import UIKit

class MenuViewController: UIViewController, UITableViewDataSource, UITableViewDelegate {
    
    
    var names = [String : AnyObject]()
    var menu = [AnyObject]()
    var menuKey = [AnyObject]()
    @IBOutlet var mainTableView : UITableView?
    
    override func viewDidLoad() {
       
        super.viewDidLoad()
        self.navigationController?.navigationBar.isHidden = true
        self.names = PGDataManager.getContentHeader() 
        self.menuKey =  self.names["key"] as! [AnyObject]
        self.menu = self.names["value"] as! [AnyObject]
        self.menuKey.append(self.menuKey[0])
        self.menu.append("Top" as AnyObject)
        self.slidingViewController().anchorRightRevealAmount = 200.0
        mainTableView?.delegate = self
        mainTableView?.dataSource = self
        mainTableView?.backgroundColor = UIColor.clear
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

    // MARK: - Table view data source

     func numberOfSections(in tableView: UITableView) -> Int {
        // #warning Incomplete implementation, return the number of sections
        return 1
    }

     func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        // #warning Incomplete implementation, return the number of rows
         return self.menuKey.count
    }
    
    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
       
        var cell = tableView.dequeueReusableCell(withIdentifier: "ListIdentifier")
        if cell == nil {
            cell = UITableViewCell(style: .default, reuseIdentifier: "ListIdentifier")
        }
        cell?.backgroundColor = UIColor.clear
        cell?.contentView.backgroundColor = UIColor.clear
        cell?.textLabel!.text  = "\(self.menu[indexPath.row])"
        cell?.textLabel?.textColor = UIColor.white
        cell?.textLabel!.font = UIFont.systemFont(ofSize: CGFloat(14))
        return cell!
    }


     func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
       
        let identifier = "\(self.menuKey[indexPath.row])"
        let navController = (self.slidingViewController().topViewController! as! UINavigationController)
        let vc = (navController.topViewController! as! PGWebViewController)
       vc.setMenuItem(identifier)
       self.slidingViewController().resetTopView(animated: true)
    }

}
